lib/android/app/src/main/java/com/reactnativenavigation/presentation/OptionsPresenter.java

```diff
@@ -34,35 +43,70 @@ private void applyTopBarOptions(TopBarOptions options) {
 
         TypefaceLoader typefaceLoader = new TypefaceLoader();
         topBar.setTitleTypeface(typefaceLoader.getTypeFace(topBar.getContext(), options.textFontFamily));
-        if (options.hidden == NavigationOptions.BooleanOptions.True) {
+        if (options.hidden == True) {
             hideTopBar(options.animateHide);
         }
-        if (options.hidden == NavigationOptions.BooleanOptions.False) {
+        if (options.hidden == False) {
             showTopBar(options.animateHide);
         }
+        if (options.collapse == True) {
```

Comment:
This logic isn't a concern of OptionsPresenter. TopBar should have a dedicated class for managing collapse

Reply:
Sure, I'll move it out as soon it's ready